### PR TITLE
Create devblog for environment differentiation

### PR DESCRIPTION
**Pull request by joellabes**
## What are you changing in this pull request and why?
Adds a new developer blog article explaining the importance of splitting out your production and CI environments if you haven't already, to take advantage of dbt Explorer and the improved Slim CI experience.

### CODE REVIEW

**Comment by joellabes** [comment]
@mani-dbt interested in your thoughts here!

**Comment by mani-dbt** [comment]
> @mani-dbt interested in your thoughts here!

@joellabes  - Thank you for putting together this content. Very helpful to understand the Slim CI & differentiating CI & production environments (especially with new CI improvement of selecting an environment instead of a job). One comment I have is that in the data landscape image, we can remove "QA Schemas," as all the target tables would be written to temporary PR schemas during this Slim CI job.

I'm also thinking that we can work on putting together another blog (different from the above) that explains how customers who want to manage DEV -> QA -> PROD deployments should configure their strategy. It would be for more advanced enterprise customers who need a QA layer in-between & NOT just dev & prod. Happy to collaborate on this with you. Attached is an image I've put together recently for a customer.
dev-qa-prod

**Comment by joellabes** [comment]
> we can remove "QA Schemas,"

Good spotting! fixed now. 

> explains how customers who want to manage DEV -> QA -> PROD deployments should configure their strategy

Yes let's do it!

File: website/blog/2023-11-06-differentiate-prod-and-staging-environments.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+- Using dedicated credentials for production runs, instead of an analytics engineer's individual dev credentials, ensures that things don't break when that long-tenured employee finally hangs up their IDE.
+
+Historically, dbt Cloud required a separate environment for _Development_, but was otherwise unopinionated in how you configured your account. This mostly just worked – as long as you didn't have anything more complex than a CI job mixed in with a couple of production jobs – because important constructs like deferral in CI and documentation were only ever tied to a single job.
+
+But as companies' dbt deployments have grown more complex, it doesn't make sense to assume that a single job is enough anymore. We need to exchange a job-oriented strategy for a more mature and scalable environment-centric view of the world. To support this, a recent change in dbt Cloud enables project administrators to mark one of their environments as the Production environment, just as has long been possible for the Development environment.
```
Comment (jasnonaz) [inline website/blog/2023-11-06-differentiate-prod-and-staging-environments.md]: Do we have docs explaining how to do this that we can link to here?

File: website/blog/2023-11-06-differentiate-prod-and-staging-environments.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+slug: differentiate-prod-and-staging-environments
+
+# image: /img/blog/2023-10-31-to-defer-or-to-clone/preview.png
+
+authors: [joel_labes]
+
+tags: [dbt Cloud]
+hide_table_of_contents: false
+
+date: 2023-11-06
```
Comment (jasnonaz) [inline website/blog/2023-11-06-differentiate-prod-and-staging-environments.md]: needs updating - lets target next tuesday?

File: website/blog/2023-11-06-differentiate-prod-and-staging-environments.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+Historically, dbt Cloud required a separate environment for _Development_, but was otherwise unopinionated in how you configured your account. This mostly just worked – as long as you didn't have anything more complex than a CI job mixed in with a couple of production jobs – because important constructs like deferral in CI and documentation were only ever tied to a single job.
+
+But as companies' dbt deployments have grown more complex, it doesn't make sense to assume that a single job is enough anymore. We need to exchange a job-oriented strategy for a more mature and scalable environment-centric view of the world. To support this, a recent change in dbt Cloud enables project administrators to mark one of their environments as the Production environment, just as has long been possible for the Development environment.
+
+Explicitly separating your Production workloads lets dbt Cloud be smarter with the metadata it creates, and is particularly important for two new features: dbt Explorer and the revised CI workflows.
+
```
Comment (jasnonaz) [inline website/blog/2023-11-06-differentiate-prod-and-staging-environments.md]: truncate

File: website/blog/2023-11-06-differentiate-prod-and-staging-environments.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+---
+
+title: Why you need to differentiate your production and staging environments in dbt Cloud
```
Comment (jasnonaz) [inline website/blog/2023-11-06-differentiate-prod-and-staging-environments.md]: I like this title but also I feel like this post is _mostly_ about why you should set a production environment and doesn't hit on staging that much. Should we add more on staging environments and / or title something that really hammers this is about the importance of a prod environment?

File: website/blog/2023-11-06-differentiate-prod-and-staging-environments.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+---
+
+:::tip The Bottom Line:
+You should split your Environments in dbt Cloud based on their purposes (e.g. Production and Staging/CI) and [set one environment as Production](/docs/deploy/deploy-environments#set-as-production-environment-beta). This will improve your CI experience and enable you to use dbt Explorer.
+:::
+
+Environmental segmentation has always been an important part of the analytics engineering workflow:
+
+- When developing new models you can [process a smaller subset of your data](/reference/dbt-jinja-functions/target#use-targetname-to-limit-data-in-dev) by using `target.name` or an environment variable.
```
Comment (jasnonaz) [inline website/blog/2023-11-06-differentiate-prod-and-staging-environments.md]: good docs linking here - should we link to the overall docs on environments somewhere?

File: website/blog/2023-11-06-differentiate-prod-and-staging-environments.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+**The old way**: [Slim CI](/guides/orchestration/set-up-ci/in-15-minutes) deferred to a single job, and would only detect changes as of that job's last build time.
+
+**The new way**: Changes are detected regardless of the job they were deployed in, removing false positives and overbuilding of models in CI.
+
+Just like dbt docs, relying on a single job to define your state for comparison purposes leads to a choice between unnecessarily rebuilding models which were deployed by another job, or creating a dedicated job that runs `dbt compile` on repeat to keep on top of all changes.
+
+By using the environment as the arbiter of state, any time a change is made to your Production deployment it will immediately be taken into consideration by subsequent Slim CI runs.
+
+## The easiest way to break apart your jobs
```
Comment (jasnonaz) [inline website/blog/2023-11-06-differentiate-prod-and-staging-environments.md]: This is great and probably the most important part of your post - would the flow work if we were to move this up to just below the intro so as not to bury the lede?

**Review by jasnonaz — CHANGES_REQUESTED** [review]
A prime example of a "we're updating our best practices based on new features" variety of dev blog - great work.

Some blocking comments on prepping the blog aspects like release date and adding a truncate and then non-blocking thoughts on wording / flow / linking out.

File: website/blog/2023-11-06-differentiate-prod-and-staging-environments.md (hunk 6 above)
Comment (joellabes) [reply website/blog/2023-11-06-differentiate-prod-and-staging-environments.md]: I've changed the link in the bottom line section up top to link directly to this section - I think rearranging to explain how without first explaining why probably isn't ideal?

**Comment by joellabes** [comment]
Changes made, i also renamed the file which makes the diff a bit funky

**Review by jasnonaz — APPROVED** [review]
🚢

**Pull request merged**